code/game/objects/effects/acid_hole.dm

```diff
@@ -41,10 +41,12 @@
 		use_wall_hole(user)
 
 
-/obj/effect/acid_hole/attack_alien(mob/living/carbon/xenomorph/user)
+/obj/effect/acid_hole/specialclick(mob/living/carbon/xenomorph/user)
```

Comment:
Do we have any guarantees the user will be a xeno and not a random carbon?

Reply:
fixed and tested

Reply:
To follow good practices, set it as what it can actually come as:
`/obj/effect/acid_hole/specialclick(mob/living/carbon/user)`
Then on the check, do an early return instead of a big nesting proc:
```
if(!isxeno(user))
	return
```

Reply:
donezie